File: C/Data Structure/Queue/program1.c
//Queue

#include<stdio.h>
#include<stdlib.h>

struct node
{
    int data;
    struct node *next;
};

typedef struct node NODE;
typedef struct node* PNODE;
typedef struct node** PPNODE;

void Enqueue(PPNODE Head, int No)
{
    PNODE newn = NULL;

    newn = (PNODE)malloc(sizeof(NODE));

    newn->data = No;
    newn->next = NULL;

    if(*Head== NULL)
    {
        *Head= newn;
    }
    else
    {
        PNODE temp = *Head;

        while(temp->next != NULL)
        {
            temp= temp->next;
        }
        temp->next= newn;
    }
}

void Dequeue(PPNODE Head)
{
    int no= 0;

    if(*Head == NULL)
    {
        printf("Queue is empty \n");
        return;
    }

    no = (*Head)->data;
    PNODE temp = *Head;
    *Head = (*Head)->next;
    free(temp);

    printf("Removed element is: %d\n", no);
}

void Display(PNODE Head)
{
    printf("Elements from Queue are :\n");

    while(Head != NULL)
    {
        printf("|%d|->", Head->data);
        Head = Head->next;
    }
    printf("NULL\n");
}

int Count(PNODE Head)
{
    int iCnt = 0;

    while(Head != NULL)
    {
        iCnt++;
        Head = Head->next;
    }
    return iCnt;
}

int main()
{
    PNODE First = NULL;
    int iRet = 0;

    Enqueue(&First, 11);
    Enqueue(&First, 21);
    Enqueue(&First, 51);
    Enqueue(&First, 101);

    Display(First);

    iRet = Count(First);
    printf("Number of elemenys in queue: %d\n",iRet);

    Dequeue(&First);
    Dequeue(&First);

    Display(First);

    iRet = Count(First);
    printf("Number of elemenys in queue: %d\n",iRet);

    return 0;
}

File: Java/Data Structure/Queue/program1.java
import java.util.*;

class node
{
    public node next;
    public int data;
}

class Queue
{
    public node head;
    public int Count;

    public Queue()
    {
        head = null;
        Count = 0;
    }

    public void Enqueue(int no)
    {
        node newn = new node();

        newn.data = no;
        newn.next = null;

        if(Count == 0)
        {
            head = newn;
        }
        else
        {
            node temp = head;

            while(temp.next != null)
            {
                temp = temp.next;
            }
            temp.next = newn;
        }

        Count++;
    }

    public void Dequeue()
    {
        if(Count == 0)
        {
            return;
        }

        node temp = head;
        head = head.next;
        temp = null;

        Count--;
    }

    public void Display()
    {
        if(Count == 0)
        {
            System.out.println("Queue is empty");
            return;
        }

        node temp = head;

        System.out.println("Elements of Queue are : ");

        while(temp != null)
        {
            System.out.print("|" +temp.data+ "| -> ");
            temp = temp.next;
        }
           System.out.println("NULL");
    }

    public int CountNode()
    {
        return Count;
    }
}

class program1
{
    public static void main(String[] args) 
    {
        Queue obj = new Queue();   

        obj.Enqueue(11);
        obj.Enqueue(21);
        obj.Enqueue(51);
        obj.Enqueue(101);

        obj.Display();

        System.out.println("Number of elements are : "+obj.CountNode());

        obj.Dequeue();
        obj.Dequeue();

        obj.Display();

        System.out.println("Number of elements are : "+obj.CountNode());
    }
}
